### PR TITLE
add python-grass.bat for Python initialization

### PR DESCRIPTION
**Pull request by landam**
Inspired by `python-qgis.bat` to allow GRASS initialization in editors like PyCharm.

### CODE REVIEW

File: mswindows/osgeo4w/python-grass.bat.tmpl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+rem #########################################################################
+rem #
+rem # GRASS initialization bat script for Python environment (OSGeo4W)
+rem #
+rem #########################################################################
+
+rem
+rem Set environmental variables
+rem
+call "%~dp0\o4w_env.bat"
```
Comment (hellik) [inline mswindows/osgeo4w/python-grass.bat.tmpl]: in python-qgis.bat, there is also

`call py3_env.bat`

as there is still also python2 living inside OSGeo4W. maybe we should get rid of python 2 at some point in OSGeo4W. python 2 is still default:

```
run o-help for a list of available commands
C:\>python
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
```

or am I overlooking something.

File: mswindows/osgeo4w/python-grass.bat.tmpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+call "%OSGEO4W_ROOT%\apps\grass\grass@POSTFIX@\etc\env.bat"
```
Comment (hellik) [inline mswindows/osgeo4w/python-grass.bat.tmpl]: related to my comment above, in _env.bat_

`call %OSGEO4W_ROOT%\bin\py3_env.bat`

ok, we have to be careful not to invoke things twice.

maybe some system refactoring needed.

File: mswindows/osgeo4w/python-grass.bat.tmpl (hunk 1 above)
Comment (landam) [reply mswindows/osgeo4w/python-grass.bat.tmpl]: Yes, but python-grass.bat calls GRASS `env.bat` file where `py3_env.bat` is introduced already.

Comment (hellik) [reply mswindows/osgeo4w/python-grass.bat.tmpl]: ok, keep it there.

**Comment by landam** [comment]
I document workflow in https://grasswiki.osgeo.org/wiki/Tools_for_Python_programming#PyCharm_IDE

@hellik You can test build no. 140

**Comment by neteler** [comment]
(note that the Wiki page has a mild overlap with https://trac.osgeo.org/grass/wiki/Submitting/Python#Editorsettingsfor4-spaceindentation)

**Comment by landam** [comment]
> is the intention of _python-grass.bat_ to GRASS python to PYTHONPATH?
> 
> as it is done in qgis.bat:
> 
> `set PYTHONPATH=%OSGEO4W_ROOT%\apps\qgis\python;%PYTHONPATH%`

done in 53116d3933bddaf33a60e6a107c28c25a7788fe4

**Comment by landam** [comment]
> any dummy example around how python-grass.bat is intended to be used?

https://grasswiki.osgeo.org/wiki/Tools_for_Python_programming#PyCharm_IDE

**Comment by hellik** [comment]
> https://grasswiki.osgeo.org/wiki/Tools_for_Python_programming#PyCharm_IDE

nice example

**Comment by hellik** [comment]
>@hellik You can test build no. 140

just overlooked this one. what should I test?

**Comment by landam** [comment]
> > @hellik You can test build no. 140
> 
> just overlooked this one. what should I test?

I thought the workflow documented at wiki, https://grasswiki.osgeo.org/wiki/Tools_for_Python_programming#PyCharm_IDE

**Comment by ecodiv** [comment]
@landam I have tried to follow the workflow mentioned above. However, Pycharm get's stuck connecting to the concole. On the console the below:

```
C:\OSGeo4W64\bin\python-grass79.bat "C:\Program Files\JetBrains\PyCharm Community Edition 2020.1\plugins\python-ce\helpers\pydev\pydevconsole.py" --mode=client --port=49336
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2020.1\plugins\python-ce\helpers\pydev\pydevconsole.py", line 5, in <module>
    from _pydev_comm.pydev_rpc import make_rpc_client, start_rpc_server, start_rpc_server_and_make_client
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2020.1\plugins\python-ce\helpers\pydev\_pydev_comm\pydev_rpc.py", line 4, in <module>
    from _pydev_comm.pydev_server import TSingleThreadedServer
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2020.1\plugins\python-ce\helpers\pydev\_pydev_comm\pydev_server.py", line 4, in <module>
    from _shaded_thriftpy.server import TServer
ModuleNotFoundError: No module named '_shaded_thriftpy'
Press any key to continue . . . 
```

**Comment by hellik** [comment]
for testing, try to start python-grass79.bat in a normal windows cmd console (not in the OSGeo4W shell:

```
C:\Users\myuser>C:\OSGeo4W64\bin\python-grass79.bat
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import grass.script as grass
>>>
```

this works here. 

now testing the pycharm workflow

**Comment by landam** [comment]
> @landam I have tried to follow the workflow mentioned above. However, Pycharm get's stuck connecting to the concole. On the console the below:

right, I am able to reproduce the problem with console. Please fill an issue. Anyway you should be able to *run* a script as described on the wiki.

**Comment by hellik** [comment]
for me the sample script is working:

```
C:\OSGeo4W64\bin\python-grass79.bat C:/Users/hkmyr/AppData/Roaming/JetBrains/PyCharmCE2020.1/scratches/test.py
{'GISDBASE': 'C:\\Users\\hkmyr\\AppData\\Local\\Temp', 'LOCATION_NAME': 'demo', 'MAPSET': 'PERMANENT'}

Process finished with exit code 0


```

**Comment by hellik** [comment]
see

https://youtrack.jetbrains.com/issue/PY-38606

**Comment by hellik** [comment]
and

https://youtrack.jetbrains.com/issues?q=ModuleNotFoundError:%20No%20module%20named%20%27_shaded_thriftpy%27

**Pull request merged**